content/zh/blog/_posts/2022-05-23-service-ip-dynamic-and-static-allocation.md

```diff
@@ -0,0 +1,268 @@
+Service across the different backing Pods.
+-->
+在 Kubernetes 中，[Services](/zh/docs/concepts/services-networking/service/)
+是一种抽象，用来暴露运行在一组 Pod 上的应用。
+Service 可以有一个集群范围的虚拟 IP 地址（使用 `type: ClusterIP` 的 Service）。
+客户端可以使用该虚拟 IP 地址进行连接， Kubernetes 为对该 Service 的访问流量提供负载均衡，以访问不同的后端 Pod。
+
+<!--
+## How Service ClusterIPs are allocated?
+-->
+## Service ClusterIP 是如何分配的？
+
+<!--
+A Service `ClusterIP` can be assigned:
+-->
+Service `ClusterIP` 有如下分配方式：
+
+<!--
+_dynamically_
+: the cluster's control plane automatically picks a free IP address from within the configured IP range for `type: ClusterIP` Services.
+-->
+**动态**
+：群集的控制平面会自动从配置的 IP 范围内为 `type:ClusterIP` 的 Service 选择一个空闲 IP 地址。
+
+<!--
+_statically_
+: you specify an IP address of your choice, from within the configured IP range for Services.
+-->
+**静态**
+：你可以指定一个来自 Service 配置的 IP 范围内的 IP 地址。
+
+<!--
+Across your whole cluster, every Service `ClusterIP` must be unique.
+Trying to create a Service with a specific `ClusterIP` that has already
+been allocated will return an error.
+-->
+在整个集群中，每个 Service 的 `ClusterIP` 必须是唯一的。
+尝试创建一个已经被分配了的 `ClusterIP` 的 Service 将会返回错误。
+
+<!--
+## Why do you need to reserve Service Cluster IPs?
+-->
+## 为什么需要预留 Service Cluster IP？
+
+<!--
+Sometimes you may want to have Services running in well-known IP addresses, so other components and
+users in the cluster can use them.
+-->
+有时，你可能希望让 Service 运行在众所周知的 IP 地址上，以便集群中的其他组件和用户可以使用它们。
+
+<!--
+The best example is the DNS Service for the cluster. Some Kubernetes installers assign the 10th address from
+the Service IP range to the DNS service. Assuming you configured your cluster with Service IP range
+10.96.0.0/16 and you want your DNS Service IP to be 10.96.0.10, you'd have to create a Service like
+this:
+-->
+最好的例子是集群的 DNS Service。一些 Kubernetes 安装程序将 Service IP 范围中的第 10 个地址分配给 DNS Service。
+假设你配置集群 Service IP 范围是 10.96.0.0/16，并且希望 DNS Service IP 为 10.96.0.10，
+那么你必须创建一个如下所示的 Service：
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    k8s-app: kube-dns
+    kubernetes.io/cluster-service: "true"
+    kubernetes.io/name: CoreDNS
+  name: kube-dns
+  namespace: kube-system
+spec:
+  clusterIP: 10.96.0.10
+  ports:
+  - name: dns
+    port: 53
+    protocol: UDP
+    targetPort: 53
+  - name: dns-tcp
+    port: 53
+    protocol: TCP
+    targetPort: 53
+  selector:
+    k8s-app: kube-dns
+  type: ClusterIP
+```
+
+<!--
+but as I explained before, the IP address 10.96.0.10 has not been reserved; if other Services are created
+before or in parallel with dynamic allocation, there is a chance they can allocate this IP, hence,
+you will not be able to create the DNS Service because it will fail with a conflict error.
+-->
+但正如我之前解释的，IP 地址 10.96.0.10 没有被保留；
+如果其他 Service 在动态分配之前创建或与动态分配并行创建，则它们有可能分配此 IP 地址，
+因此，你将无法创建 DNS Service，因为它将因冲突错误而失败。
+
+<!--
+## How can you avoid Service ClusterIP conflicts? {#avoid-ClusterIP-conflict}
+-->
+## 如何避免 Service ClusterIP 冲突？
```

Comment:
Missing `{#anchor}`

Ref: [Localization Guild](https://kubernetes.io/zh/docs/contribute/localization_zh/#%E5%85%B3%E4%BA%8E%E9%93%BE%E6%8E%A5)